### PR TITLE
tests: Coverage onboarding assembly

### LINKED ISSUE #934: Coverage onboarding assembly
Improve test coverage of the tutorial feature

### PR DESCRIPTION
**Pull request by sandrade-dcl**
## What does this PR change?
Fixes #934 

- [x] Refactor `TutorialController` to make it a non-MonoBehaviour class.
- [x] Create the corresponding tests to coverage all the `onboarding` assembly.

## Coverage resulting from this PR
https://renderer-artifacts.decentraland.org/branch-coverage/tests/Coverage-onboarding-assembly

## Test link
https://play.decentraland.zone/?renderer-branch=tests/Coverage-onboarding-assembly

## Our Code Review Standards
https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

**Review by pravusjif — COMMENTED** [review]
Testing as guest I never get the "Alice Tutorial" have we intentionally removed it?

File: unity-renderer/Assets/Tutorial/Scripts/TutorialConfigurator.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
+using Cinemachine;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace DCL.Tutorial
+{
+    public class TutorialConfigurator : MonoBehaviour
```
Comment (BrianAmadori) [inline unity-renderer/Assets/Tutorial/Scripts/TutorialConfigurator.cs:7]: Not sure if you already did it, but we can use this opportunity to remove the `TutorialController` from `InitialScene`.

File: unity-renderer/Assets/Tutorial/Scripts/TutorialConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
+using Cinemachine;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace DCL.Tutorial
+{
+    [CreateAssetMenu(fileName = "TutorialConfiguration", menuName = "TutorialConfiguration")]
+    public class TutorialConfiguration : ScriptableObject
```
Comment (BrianAmadori) [inline unity-renderer/Assets/Tutorial/Scripts/TutorialConfiguration.cs:9]:
```suggestion
    public class TutorialSettings : ScriptableObject
```
I think it's better if we use the word `settings` here, as it's widely used through the project for the cases where we need to inject an object with configuration.

File: unity-renderer/Assets/Tutorial/Scripts/TutorialController.cs [hunk 3]
```diff
@@ -580,14 +527,19 @@ private void MoreMenu_OnRestartTutorial()
 
         private bool IsPlayerInsideGenesisPlaza()
```
Comment (BrianAmadori) [inline unity-renderer/Assets/Tutorial/Scripts/TutorialController.cs:528]: I think this method can be made pure and `static` and be unit tested if we just inject the `world` and needed dependencies.

**Review by BrianAmadori — COMMENTED** [review]
Just left some suggestions, lmk wdyt about them.

File: unity-renderer/Assets/Tutorial/Scripts/TutorialConfigurator.cs (hunk 1 above)
Comment (sandrade-dcl) [reply unity-renderer/Assets/Tutorial/Scripts/TutorialConfigurator.cs]: Done in https://github.com/decentraland/unity-renderer/pull/966/commits/64e96180bad0746ca4f7146eada3252b14aed9b4. Thanks for the feedback!

File: unity-renderer/Assets/Tutorial/Scripts/TutorialConfiguration.cs (hunk 2 above)
Comment (sandrade-dcl) [reply unity-renderer/Assets/Tutorial/Scripts/TutorialConfiguration.cs]: You're right. Done in https://github.com/decentraland/unity-renderer/pull/966/commits/989912d60c62b7549e7b7a24639991144b88cbbc. Thanks!

**Comment by sandrade-dcl** [comment]
> Testing as guest I never get the "Alice Tutorial" have we intentionally removed it?

Fixed! Good catch! Thanks :)

**Review by pravusjif — APPROVED** [review]

**Review by BrianAmadori — APPROVED** [review]

**Pull request merged**